nixos/modules/programs/fish.nix

```diff
@@ -107,9 +107,19 @@ in
       # This happens before $__fish_datadir/config.fish sets fish_function_path, so it is currently
       # unset. We set it and then completely erase it, leaving its configuration to $__fish_datadir/config.fish
       set fish_function_path ${pkgs.fish-foreign-env}/share/fish-foreign-env/functions $__fish_datadir/functions
-      
+
+
+      # Here we prepare to filter out warnings about non-existent $PATH entries
+      # because NixOS uses them intentionally.
+      set -l warnings_filter_pattern '^set: Warning: \$PATH entry ".*" is not valid \(No such file or directory\)$'
+
+      # `fenv` should not attempt to modify $_, but it seems to do so when
+      # used to run the bash `source` builtin. We don't need to hear about it,
+      # either.
+      set warnings_filter_pattern $warnings_filter_pattern"|"'|^set: Tried to change the read-only variable “_”$'
```

Comment:
I think something like that would work, if only because fenv doesn't correctly detect and address erased variables. If fenv did that, it would detect a change and still attempt to set (because $_ is always see to something initially). Should we go for it anyway?